### ISSUE TITLE
Expose more SSH connection features in "Connect to GNU gdb via ssh"

### ISSUE THREAD

**Issue by NeutralKaon** [NationalSecurityAgency/ghidra#3474]
Thank you for releasing Ghidra. I have found it very useful in the context of my work as an academic medical physicist. 

**Is your feature request related to a problem? Please describe.**
The debugging feature in Ghidra is great, and the opportunity to connect to gdb via ssh is very useful. However, often remote hosts are behind several layers of firewalls. I have a large and fairly complex `~/.ssh/config` set up to deal with this. Ghidra reads SSH keys from `~/.ssh`, but not the config file. I understand that this is a complex thing to do, but it would be a significant quality-of-life improvement to expose more of ssh's (myriad) functionality. For example, the underlying `ch.ethz.ssh2` library used in `Ghidra/Debug/Debugger-agent-gdb/src/main/java/agent/gdb/pty/ssh/GhidraSshPtyFactory.java` [natively supports](http://www.ganymed.ethz.ch/ssh2/javadoc/ch/ethz/ssh2/Connection.html#setProxyData(ch.ethz.ssh2.ProxyData)) the creation and destruction of port forwarding. I am aware that you are aware of the limitations of this module -- it's mentioned in the debugger help. 

**Describe the solution you'd like**
As a very minor quality-of-life improvement, it would be nice if Ghidra exposed more SSH options, read commands such as `ProxyJump` or `ProxyCommand` from a user's `~/.ssh/config` and implemented them in this dialogue. I accept that this is probably a low priority request. Alternatively, consider this an upvote for the "please improve ssh functionality"! 

**Describe alternatives you've considered**
Obviously, in this case, a user can manually create a locally listening tunnel and redirect the gdb interface to that. It becomes harder, however, if e.g. a nonstandard cipher is used or other features that openSSH provides are relied upon. 

**Additional context**
I do not believe there is any context that is required, but just to be clear it is this dialogue I am talking about: 

![image](https://user-images.githubusercontent.com/18312003/135998811-dc5ee0e8-a281-44b1-8593-f6981e4b605d.png)

**Comment by d-millar** [comment]
Apologies for the slow response - have added a ticket.

**Comment by nsadeveloper789** [comment]
Agreed, loading/using the SSH config options would be ideal. I think we should convert to JSch before we do that, though. However, we can take this as cause for converting sooner :)
